modules/javafx.graphics/src/test/java/test/javafx/scene/CssStyleHelperTest.java

```diff
@@ -684,6 +689,33 @@ public void shouldDetectSimpleInfiniteLoop() throws IOException {
         root.getChildren().addAll(a);
 
         assertDoesNotThrow(() -> stage.show());  // This should not result in a StackOverflowError
+        assertEquals(1, CssParser.errorsProperty().size());
+
+        ParseError error = CssParser.errorsProperty().getFirst();
+
+        assertEquals(PropertySetError.class, error.getClass());
+
+        // Note: on Windows, the message is using inconsistent line endings (sometimes Windows, sometimes Linux)
+        // so I've stripped it.
```

Comment:
would looking for substrings be a better solution?  for example:

```
contains("Caught java.lang.IllegalArgumentException: Loop detected in *.root{") &&
contains("while resolving '-fx-base'' while calculating value for '-fx-background-color' from rule '*.pane' in stylesheet"
```

Reply:
Yeah, this seems less fragile. If we ever run into a problem because the message changes slightly, we could consider such a change in the future.

Reply:
I'm not sure what way is being advocated.  It seems both of you favor a contains check?

For me, I always verify strings intended for user consumption completely so that any change would fail a test.  If the change is intended (like fixing the missing line feeds), then the string can be fixed at the same time.  With the complete string, the test also serves as a way to see the full output (which in this case, is a bit abysmal, so perhaps we should follow up with a fix).

Reply:
> It seems both of you favor a contains check?

Yeah, that is what I was saying, but I don't mind either way. And you raise a good point about the advantage of having any change to the string cause a failure (as long as the string isn't prone to spurious changes).

So I have no objection to your leaving it as is.